core/src/main/java/org/apache/iceberg/jdbc/JdbcTableOperations.java

```diff
@@ -71,7 +68,7 @@ public void doRefresh() {
     Map<String, String> table;
 
     try {
-      table = getTable();
+      table = JdbcUtil.tableOrView(true, connections, catalogName, tableIdentifier);
```

Comment:
This doesn't look correct.  We should only be loading table in the table operations or failing if it is a view.  In the unlikely event someone drops a table and creates a view with the same identifier, we shouldn't attempt to load that as a table.

Reply:
Now I'm seeing what this is doing.  I think we need to update this method as it's rather confusing that the first param is toggling between loading a table and a view.  We should either split it to to be `loadTable` and `loadView` or have something more explicit (like an enum) for the type.

Reply:
It's check only table here (as the first argument to `tableOrView()` method is `true`. I agree that it doesn't seem obvious at first glance, so I will do specific methods for view and table to avoid confusion.